### PR TITLE
avahi dependency support

### PR DESCRIPTION
**Pull request by daminetreg**
Hi @ruslo ,

Continuing my PR collection : #227 #233 #234. :twisted_rightwards_arrows:

Finally a working Avahi integration thanks to your nice feedback and extensions to hunter_autotools_project.

I hope you won't get any missing dependencies, I checked and all libs (up to the optional gobject support one that I sadly) were found and taken from hunter install dir.

So normally it should always compile even on a naked system.

Cheers,

### CODE REVIEW

**Comment by ruslo** [comment]
> I hope you won't get any missing dependencies

@daminetreg unfortunately it's not, you can test it with dbus uninstalled:
```bash
> sudo apt-get remove libdbus-1-dev
> PROJECT_DIR=examples/Avahi TOOLCHAIN=gcc ./jenkins.py --verbose
checking for DBUS... no
configure: error: Package requirements ( dbus-1 >= 0.34 ) were not met:

No package 'dbus-1' found
```

It trying to find `*.pc` file but there is no such file installed:
```
Install the project...
cmake -P cmake_install.cmake
-- Install configuration: "Debug"
-- Install configuration: "Debug"
-- Install configuration: "Debug"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/lib/libdbus-1d.so.3.14.3
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/lib/libdbus-1d.so.3
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/lib/libdbus-1d.so
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/lib/libdbus-1d.so.3.14.3"
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-address.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-bus.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-connection.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-errors.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-macros.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-memory.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-message.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-misc.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-pending-call.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-protocol.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-server.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-shared.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-signature.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-syntax.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-threads.h
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-types.h
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/include/dbus/dbus-arch-deps.h
-- Install configuration: "Debug"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/share/dbus-1/session.conf
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/etc/dbus-1/session.conf
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/share/dbus-1/session.d
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/share/dbus-1/services
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/share/dbus-1/system.conf
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/etc/dbus-1/system.conf
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/share/dbus-1/system.d
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/share/dbus-1/system-services
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/var/run/dbus
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/lib/dbus-daemon-launch-helper
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/lib/dbus-daemon-launch-helper"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-daemon
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-daemon"
-- Install configuration: "Debug"
-- Install configuration: "Debug"
-- Install configuration: "Debug"
-- Up-to-date: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/var/lib/dbus
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-test-tool
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-test-tool"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-update-activation-environment
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-update-activation-environment"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-monitor
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-monitor"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-launch
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-launch"
-- Installing: /.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-send
-- Removed runtime path from "/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Install/bin/dbus-send"
-- Install configuration: "Debug"
make[3]: Leaving directory `/.../hunter/_testing/Hunter/_Base/5a16c0a/a27ed63/b270781/Build/dbus/Build/dbus-Debug-prefix/src/dbus-Debug-build'
```

I've tried to set:
* `DBUS_CFLAGS="-I/.../hunter/_testing/Hunter/_Base/aa315d0/a27ed63/b270781/Install/include"`
* `DBUS_LIBS="-L/.../hunter/_testing/Hunter/_Base/aa315d0/a27ed63/b270781/Install/lib -ldbus-1"`

as error message suggest
```
Alternatively, you may set the environment variables DBUS_CFLAGS
and DBUS_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

but got new error:
```
checking for DBUS... yes
Package dbus-1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `dbus-1.pc'
to the PKG_CONFIG_PATH environment variable
No package 'dbus-1' found
Error: Couldn't determine the version of your D-Bus package.
  This is probably an error in this script, please report it
  along with the following information:
      Base D-Buss version =''
      DBUS_VERSION_MAJOR='0'
      DBUS_VERSION_MINOR='0'
      DBUS_VERSION_MICRO='0'
```

**Comment by daminetreg** [comment]
Oh sorry, I did not do my job quite well in checking this.

This was due to the cmake build of dbus which wasn't configure/installing the pkgconfig file that autoreconf uses to find dbus. Now the cmake build does is also and I could build avahi with dbus deinstalled.

I noticed that it was also taking glib from my system, so I added the --disable-glib --disable-gobject call.

So now it should work :smile:.

File: examples/Avahi/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Emulate HunterGate:
+# * https://github.com/hunter-packages/gate
+include("../common.cmake")
+
+project(download-Avahi)
+
+hunter_add_package(Avahi)
+find_package(Avahi REQUIRED)
+include_directories(AFTER ${Avahi_INCLUDE_DIRS})
```
Comment (ruslo) [inline examples/Avahi/CMakeLists.txt]: Not needed since we have `INTERFACE_INCLUDE_DIRECTORIES` property set.

**Comment by ruslo** [comment]
@daminetreg Please set PR's destination branch to `ruslo:develop`

> So now it should work :smile:

Yep, it works now, thanks!

**Pull request closed without merging**